Refactor access derivative creation script

kfrn opened the pull request:
More refactoring! I thought I'd try to streamline things a bit - let me know what you think.

~**Edit** - okay I just noticed I made a few mistakes. Opps. I can fix them in the next day or so, or anyone else can go ahead.~ Now fixed ... hopefully!

kfrn committed:
UX: reorder user queries

Now the in & out-point queries come directly after the "Do you want to trim" query

kfrn committed:
Add gitignore

kfrn committed:
Refactor: extract ffmpeg command generation into separate function

kfrn commented on main.py:
```diff
@@ -55,95 +54,81 @@ def main():
+    ]
+
+    stream_settings = [
+        # TODO "setdar=dar={0}".format(dar),   # set DAR to aspect ratio
```
Not sure what's going on with `setdar` - when I had this in the script, I was getting the error
```bash
Unable to find a suitable output format for 'setdar=dar=1.3333333333333333'
setdar=dar=1.3333333333333333: Invalid argument
```
The ffmpeg [docs](https://ffmpeg.org/ffmpeg-filters.html#Examples-97) give an example of `setdar=dar=1.77777`, so it could be too many decimal places? Or a string v. float thing. Anyway, I leave this to later to debug :)

kfrn replied:
I realised what the deal is here - `setdar` is not a standalone thing, it's part of the video filter. Will fix in a follow-up commit

Warblefly replied:
If it's of any use (and I've not really looked at the rest of this thread) you can use ratios e.g. setdar=4/3

kfrn replied:
Set the DAR in e252abc. If there's an overlay (watermark), that needs to be part of the same filtergraph, so that commit includes logic to combine them if necessary

kfrn replied:
I just saw your comment @Warblefly! I hadn't refreshed the page. Yep, both floats (`1.333(3)`) and ratios (`4:3`) work here. What I was doing wrong was that I forgot that `setdar` was a filter, not a top-level command (even when ffmpeg was telling me just that!), opps.  
Changed in e252abc - would love feedback if you have it!

kfrn commented on main.py:
```diff
@@ -55,95 +54,81 @@ def main():
+
+    watermark_settings = [
+        "-i", watermark_file,
+        "-filter_complex", "[0]yadif=0:-1:1[v];[v][1]overlay"  # overlay filter for supplied png
```
Ah sorry, just realised that I copied the `yadif` in here by accident. Will fix later on!

kfrn replied:
Opps - just saw some of the Twitter discussion. It looks like you can't have `-filter_complex` AND `-vf`? Another thing for me to fix ...

kfrn replied:
Changed to simply `overlay` in 5b2e721 :smile:

kfrn commented on main.py:
```diff
@@ -55,95 +54,81 @@ def main():
+        # TODO "setdar=dar={0}".format(dar),   # set DAR to aspect ratio
+        "-c:v",     "libx264",          # encode video stream as H.264
+        "-pix_fmt", "yuv420p",          # 4:2:0 chroma subsubsampling
+        "-vf",      "yadif",            # deinterlace video stream
```
As above - should probably be `"[0]yadif=0:-1:1[v]"`. (Need to refresh my memory on what that means ...)

**Edit** Actually, I'd be tempted to have deinterlacing as a user-set option

kfrn replied:
Removed deinterlacing for the moment in e252abc

Like @kieranjol [suggested](https://twitter.com/kieranjol/status/1171756604544430081), it's probably better as an opt-in thing.

digitensions replied:
Agreed, in light of a comment I received on FFmpeg-user from Carl Eugen also, there's no guarantee the metadata for the asset is providing the correct interlacing data - particularly in MACE's very variable back catalogue of weird files...  Sounds great to create an opt in :)

digitensions replied:
> As above - should probably be `"[0]yadif=0:-1:1[v]"`. (Need to refresh my memory on what that means ...)
 
If it helps, this is the conversation from ffmpeg-user that influenced the decision:
http://ffmpeg.org/pipermail/ffmpeg-user/2019-September/045365.html

kfrn replied:
CEH, ah yes I remember him, `Complete, uncut console output missing` 😁

Cheers for the link :-)

kfrn commented on main.py:
```diff
@@ -55,95 +54,81 @@ def main():
+
+    output_settings = [
+        "-metadata", "copyright='Media Archive for Central England'",
+        "-metadata", "comment='Please contact MACE to license this footage on 01522 837750'",
```
TODO: unquote the copyright and comment messages - oops!

kfrn replied:
Fixed in 5b2e721 :smile:

kfrn committed:
Refactor: construct filtergraph to set DAR and (potentially) watermark

You can't use both a video filter (`-vf`) and a complex filtergraph (`-filter_complex`) in the same command.
We have two filters to apply:
* set the DAR
* set the overlay (watermark) - user option

If doing both, we need to include both actions in a single filtergraph - so I've extracted that into a standalone function.

kfrn commented on main.py:
```diff
@@ -55,95 +54,86 @@ def main():
+
+    watermark_settings = [
+        "-i",               watermark_file,
+        "-filter_complex", "setdar=dar={0},overlay".format(dar)
```
Previously the command was `"[0]yadif=0:-1:1[v];[v][1]overlay"`. Presumably there is a similar/more sophisticated way to include both `setdar` and `overlay` in the filtergraph, but this seems to work, and I believe in keeping it simple where possible :laughing:

digitensions commented:
Thanks Katherine! 👍 I'll merge and run some tests!
